ocs_sample_library_preview/SharedStreams.py

```diff
@@ -0,0 +1,483 @@
+from typing import Any
+
+from .BaseClient import BaseClient
+from .SDS.SdsResultPage import SdsResultPage
+from .SDS.SdsStream import SdsStream
+from .SDS.SdsType import SdsType
+from .PatchableSecurable import PatchableSecurable
+
+from .Streams import Streams
+
+class SharedStreams(PatchableSecurable, object):
```

Comment:
I am not sure that this needs to be a new class. Is there a reason that we can't use the existing ```Streams``` class where the community Id and tenant Id are optional? I can see this becoming difficult to maintain as additional features are added to communities as we will essentially have double the work.

Reply:
I see your point here, that was an option I considered but ultimately went this way instead. The biggest reason was to make it as clear as possible for users, I personally don't like optional parameters that drive logic as they can make the function confusing or be misinterpreted, and I felt that this would communicate intent more clearly.

I also didn't want to overload the Streams class more since that class is already very big. This is of course just my opinion but I prefer having smaller more cohesive classes/methods when possible. 

As for maintenance I actually believed that this will be easier to maintain since it separates the logic from the Streams class without needing to duplicate code for most methods. This way if we introduce further logic and features for communities or streams, they can grow separately while sharing common logic. The thought is for this pattern to be used for other features as well, creating clients such as SharedDataviews, SharedAssets, etc. moving forward.

That's just my thoughts on it but please let me know what you think, I'd be happy to make any changes we deem best. We could always jump on a call as well :)

Reply:
After reading your comment and re-reading the class, I am convinced. There is minimal duplication of code and your method keeps the class sizes smaller, which I agree is better.